src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -0,0 +1,34 @@
+package com.kodilla.ecommercee.domain;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+
+import javax.persistence.*;
+import java.util.List;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Entity(name = "CARTS")
+public class Cart {
+
+    @Id
+    @GeneratedValue
+    @NotNull
+    @Column(name = "CART_ID", unique = true)
+    private Long cartId;
+
+    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinColumn(name = "USER_ID")
+    private User userId;
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(
+            name = "PRODUCTS_IN_CARTS",
+            joinColumns = {@JoinColumn(name = "CART_ID", referencedColumnName = "CART_ID")},
+            inverseJoinColumns = {@JoinColumn(name = "PRODUCT_ID", referencedColumnName = "PRODUCT_ID")}
+    )
+    private List<Product> products;
```

Comment:
I still think that it should be a @OneToMany relationship with the ProductsInCart entity.

Reply:
I disagree, in the diagram it says ProductsInCart is a result table of the M:N relationship. I don't think there needs to be a separate entity for that.

Reply:
As agreed in meeting i added ProductsInCart and remade the relations.